### DISCUSSION TITLE
How to modify the parameters in the MESH in Model

### DISCUSSION THREAD

**Discussion by suzhouhu** [StereoKit/StereoKit#376] [Q&A]
I used Stereokit to develop a molecular simulator. In short, it is based on the space coordinates of an atom (small ball) in the molecule, and the chemical bonds (cylindrical) connected between related atoms.
```csharp
_BondMes = Mesh.GenerateCylinder (bonddiameter, (xy + convert.tosingle (atomutils.getradim (at)) * Atomdiameter);
mdl.addnode (
    a + "-" + b,
    Matrix.t (MDDLPNT),
    _bondmes,
    Bondparameters);
```
Wait for the construction of the molecular structure smoothly. However, I hope that the size of the chemical key (cylindrical) by clicking the relevant button is the size of the Bonddiameter.
I noticed that the color of the chemical bonds can be modified in the following ways:
```csharp
Foreach (var node in mdl.visuals.where (n => n.name.contains ("c"))
{{
    node.mesh = node.mesh.copy ();
    node.matial.transparence = transparency.add;
    node.matial.depthwrite = false;
}
```
However, it is impossible to find a modification  "BondDiameter"

**Comment by maluoi** [comment]
`Mesh.GenerateCylinder` does not create a high-level object that understands the concept of Cylinder, it creates a `Mesh`, which is always just a collection of vertices and indices. It's just triangles at this point!

You might be interested in using scale like this to achieve the same effect though:
```csharp
Vec3 scale = new Vec3(
    BondDiameter,
    (xy + convert.ToSingle(atomUtils.GetRadim(at)) * AtomDiameter
    BondDiameter);

Mesh _BondMesh = Mesh.GenerateCylinder(1, 1);

mdl.AddNode (
    a + "-" + b,
    Matrix.TS(MDDLPNT, scale),
    _BondMesh,
    BondParameters);
```
You can then update the `ModelNode`'s `LocalTransform` with a new XZ scale!

**Reply by suzhouhu** [reply]
As you said, I modified my code, but still don't know how to assign the size of the cylinder:

```csharp
Vec3 scale = new(BondDiameter, BondDiameter, xy / 2);
_bondMes = Mesh.GenerateCylinder(1, 1, Vec3.Forward);
mdl.AddNode(
    "C1_C2",
    Matrix.TRS(mddlPnt2, Quat.LookDir(bndDir), scale),
    _bondMes,
    BondParameters );
if (UI.Button("球棍"))
    foreach (var node in mdl.Visuals.Where(n => n.Name.Contains("C")))
    {
        // error# CS1612 The return value of 'Matrix.Scale' cannot be modified because it is not a variable
        node.LocalTransform.Scale.XY=(.1f,.1f);     
    }
```

`node.LocalTransform.Scale.XY=(.1f,.1f);` This code is wrong, how to modify it?

**Reply by maluoi** [reply]
On a `Matrix`, `Scale` is read-only! You need to create a new `Matrix` here.

```csharp
node.LocalTransform.Decompose(out Vec3 pos, out Quat rot, out Vec3 scale);
scale.XY = new Vec2(.1f, .1f);
node.LocalTransform = Matris.TRS(pos, rot, scale);
```

**Reply by suzhouhu** [reply]
Very good! Solved my big problem. Thank you so much for your enthusiasm
